File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    i = 0
    x = 1
    nx = 0
    
    while(len(s) > 0):
        i += 1

        if i > len(s)-1:
            answer += 1
            break
        
        if s[0] == s[i]:    x += 1
        else:               nx += 1
        
        if x == nx:
            answer += 1
            s = s[i+1:]
            i = 0
            x = 1
            nx = 0
    
    return answer

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.cs
using System;

public class Solution {
    public int solution(string s) {
        int answer = 0;
        while(s.Length > 0){
            int sameCnt = 0, diffCnt = 0;
            char same = s[0];
            foreach(char spel in s){
                if(spel == same)
                    sameCnt++;
                else
                    diffCnt++;
                if(sameCnt == diffCnt)
                    break;
            }
            s = s.Substring(sameCnt + diffCnt);
            answer++;
        }
        return answer;
    }
}